### PR TITLE
Merge fullRefund and expireFor

### LINKED ISSUE #5954: Merge fullRefund and expireKeyFor
**Describe the bug**
`expireKeyFor` is the same as a `fullRefund` with 0 refunded.

**Expected behavior**
We can merge these into a single function to simplify our API/ABI.  When we do we may want to select a new name for the function as neither expireKeyFor or fullRefund describe the feature well.

### PR DESCRIPTION
**Pull request by NickCuso**
# Description

Merge two similar functions into one.

# Issues

Fixes #5954
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/contracts/mixins/MixinLockCore.sol [hunk 1]
```diff
@@ -107,7 +107,7 @@ contract MixinLockCore is
    * considering the available balance. Set to 0 or MAX_UINT to withdraw everything.
    *
    * TODO: consider allowing anybody to trigger this as long as it goes to owner anyway?
-   *  -- however be wary of draining funds as it breaks the `cancelAndRefund` and `fullRefund`
+   *  -- however be wary of draining funds as it breaks the `cancelAndRefund` and `fullexpireAndRefundForRefund`
```
Comment (nfurfaro) [inline smart-contracts/contracts/mixins/MixinLockCore.sol]: `fullexpireAndRefundForRefund` needs an edit. should be `expireAndRefundFor` right?

File: smart-contracts/contracts/mixins/MixinKeys.sol [hunk 2]
```diff
@@ -89,21 +89,6 @@ contract MixinKeys is
...
-  function expireKeyFor(
-    address _owner
-  )
-    public
-    onlyOwner
-    hasValidKey(_owner)
-  {
-    Key storage key = keyByOwner[_owner];
-    key.expirationTimestamp = block.timestamp; // Effectively expiring the key
-    emit ExpireKey(key.tokenId);
```
Comment (nfurfaro) [inline smart-contracts/contracts/mixins/MixinKeys.sol]: I guess the `CancelKey` event in `_cancelAndRefund` replaces this one.

File: smart-contracts/contracts/mixins/MixinLockCore.sol (hunk 1 above)
Comment (NickCuso) [reply smart-contracts/contracts/mixins/MixinLockCore.sol]: ah whoops, fixed

File: smart-contracts/contracts/mixins/MixinKeys.sol (hunk 2 above)
Comment (NickCuso) [reply smart-contracts/contracts/mixins/MixinKeys.sol]: It's still in use by `shareKey` today

**Pull request merged**